public/js/clients/firefox/events.js

```diff
@@ -59,19 +65,18 @@ function resumed(_, packet) {
 function newSource(_, packet) {
   const { source } = packet;
 
-  if (NEW_SOURCE_IGNORED_URLS.indexOf(source.url) > -1) {
-    return;
-  }
-
-  if (source.introductionType == "debugger eval") {
-    return;
+  source.hidden = false;
+  if (isHidden(source)) {
+    source.url = `evals/console-evauation-${evalIndex++}.js`;
+    source.hidden = true;
```

Comment:
Yeah, I see where you are going with this. But I think there are some problems with this. If it's a source we should ignore, that doesn't mean it's a console evaluation, so the URL is a misnomer.

There are several types of sources, and some of them we should just flat out ignore. If you're writing tests and expect one of these sources to be in the store, that's a serious problem and we should figure out why it's not in the store instead of putting it in the store. And sources in `NEW_SOURCE_IGNORED_URLS` should be ignored and never added to the store. Apparently a source with `introductionType` of `debugger eval` should be as well. That is code that the debugger frontend sends to the server to eval, and should never exist in the debugger itself.

What you're probably looking for is eval sources. These are introduced by `eval` in the debuggee and don't have a URL, unless they have been given one with the `//# sourceURL` pragma. If they don't have a URL, they should still exist in the store but the frontend should ignore them. So they are different than the above sources: the debuggee is concerned with them, but there is no good way to show non-URL sources easily.

The problem with `NEW_SOURCE_IGNORED_INTRODUCTION_TYPES` is you've conflated `debugger eval` and just `eval`. The former we should just ignore (never add to the store) and the latter we should add but the sources tree should ignore it if it doesn't have a URL. Your changes here would ignore valid eval-based sources that are given a URL with the `sourceURL` pragma, which currently should work fine.

This is how it works currently though. We ignore the right sources, and then the sources tree just ignores non-URL sources (https://github.com/devtools-html/debugger.html/blob/master/public/js/utils/sources-tree.js#L36). So dig into your tests and figure out what sources it's expected to exist, and go from there. That's where the real bug is.